compiler/src/iree/compiler/Codegen/Utils/Utils.cpp

```diff
@@ -1175,9 +1176,25 @@ computeDimUpperBound(Value shapedValue, unsigned dimNum,
           shapedValue, dimNum,
           /*vscaleMin=*/vscaleRange->min,
           /*vscaleMax=*/vscaleRange->max, presburger::BoundType::UB);
-  if (succeeded(maybeDimBound))
-    return maybeDimBound->getSize();
-  return failure();
+  if (failed(maybeDimBound))
+    return failure();
+  auto boundSize = maybeDimBound->getSize();
+  if (succeeded(boundSize))
+    return boundSize;
+  if (roundUp == RoundUpVscaleMultiple::No)
+    return failure();
+  // If the upper bound map is of the form `add(subExpr, cst)` (cst <= 0),
+  // round it up to `subExpr`, which is _likely_ to be a vscale multiple.
```

Comment:
Which expression in your test would this apply to?

Reply:
It's not something you see in the test source. It's the expression computed by value bounds analysis for the scalable upper bound of a dim.

Reply:
> , which is _likely_ to be a vscale multiple

I'd skip this - it implies that this logic is specific to `vscale` (while it isn't).